Execute Async Fn in paralle/code.js

```diff
@@ -0,0 +1,24 @@
+/**
+ * @param {Array<Function>} functions
+ * @return {Promise<any>}
+ */
+var promiseAll = function (functions) {
+    return new Promise((resolve, reject) => {
+        const res = new Array(functions.length);
```

Comment:
i know here you have constraint of functions array always have atleast one element. but you should handle the scenario for empty array passed as well.